### ISSUE TITLE
java.lang.ClassNotFoundException: net.minecraft.server.v1_18_R1.NBTTagCompound

### ISSUE THREAD

**Issue by CamoMano** [elian1203/ezAuctions#16]
The plugin does not appear to show the item. Likely due to something being different with 1.18.

```
[Thu, 2. Dec 2021 03:17:34 UTC INFO] CamoMano issued server command: /auction start 1 1
[Thu, 2. Dec 2021 03:17:34 UTC WARN] Error getting xp needed for repair for BONE
[Thu, 2. Dec 2021 03:17:34 UTC WARN] Error getting minecraft name for BONE
[Thu, 2. Dec 2021 03:17:34 UTC WARN] java.lang.ClassNotFoundException: net.minecraft.server.v1_18_R1.NBTTagCompound
[Thu, 2. Dec 2021 03:17:34 UTC WARN] at org.bukkit.plugin.java.PluginClassLoader.loadClass0(PluginClassLoader.java:151)
[Thu, 2. Dec 2021 03:17:34 UTC WARN] at org.bukkit.plugin.java.PluginClassLoader.loadClass(PluginClassLoader.java:103)
[Thu, 2. Dec 2021 03:17:34 UTC WARN] at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
[Thu, 2. Dec 2021 03:17:34 UTC WARN] at java.base/java.lang.Class.forName0(Native Method)
[Thu, 2. Dec 2021 03:17:34 UTC WARN] at java.base/java.lang.Class.forName(Class.java:375)
[Thu, 2. Dec 2021 03:17:34 UTC WARN] at ezAuctions-1.6.4-SNAPSHOT.jar//net.urbanmc.ezauctions.util.ReflectionUtil.getNMSClass(ReflectionUtil.java:179)
[Thu, 2. Dec 2021 03:17:34 UTC WARN] at ezAuctions-1.6.4-SNAPSHOT.jar//net.urbanmc.ezauctions.util.ReflectionUtil.getNBTTagCompoundClass(ReflectionUtil.java:171)
[Thu, 2. Dec 2021 03:17:34 UTC WARN] at ezAuctions-1.6.4-SNAPSHOT.jar//net.urbanmc.ezauctions.util.ReflectionUtil.getItemAsJson(ReflectionUtil.java:103)
[Thu, 2. Dec 2021 03:17:34 UTC WARN] at ezAuctions-1.6.4-SNAPSHOT.jar//net.urbanmc.ezauctions.object.Auction.formatMessage(Auction.java:277)
[Thu, 2. Dec 2021 03:17:34 UTC WARN] at ezAuctions-1.6.4-SNAPSHOT.jar//net.urbanmc.ezauctions.object.Auction.getStartingMessage(Auction.java:212)
[Thu, 2. Dec 2021 03:17:34 UTC WARN] at ezAuctions-1.6.4-SNAPSHOT.jar//net.urbanmc.ezauctions.runnable.AuctionRunnable.broadcastStart(AuctionRunnable.java:59)
[Thu, 2. Dec 2021 03:17:34 UTC WARN] at ezAuctions-1.6.4-SNAPSHOT.jar//net.urbanmc.ezauctions.runnable.AuctionRunnable.<init>(AuctionRunnable.java:36)
[Thu, 2. Dec 2021 03:17:34 UTC WARN] at ezAuctions-1.6.4-SNAPSHOT.jar//net.urbanmc.ezauctions.manager.AuctionManager.addToQueue(AuctionManager.java:44)
[Thu, 2. Dec 2021 03:17:39 UTC WARN] at ezAuctions-1.6.4-SNAPSHOT.jar//net.urbanmc.ezauctions.command.AuctionCommand.startAuction(AuctionCommand.java:436)
[Thu, 2. Dec 2021 03:17:39 UTC WARN] at ezAuctions-1.6.4-SNAPSHOT.jar//net.urbanmc.ezauctions.command.AuctionCommand.start(AuctionCommand.java:346)
[Thu, 2. Dec 2021 03:17:39 UTC WARN] at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[Thu, 2. Dec 2021 03:17:39 UTC WARN] at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[Thu, 2. Dec 2021 03:17:39 UTC WARN] at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[Thu, 2. Dec 2021 03:17:39 UTC WARN] at java.base/java.lang.reflect.Method.invoke(Method.java:568)
[Thu, 2. Dec 2021 03:17:39 UTC WARN] at ezAuctions-1.6.4-SNAPSHOT.jar//net.urbanmc.ezauctions.acf.RegisteredCommand.invoke(RegisteredCommand.java:152)
[Thu, 2. Dec 2021 03:17:39 UTC WARN] at ezAuctions-1.6.4-SNAPSHOT.jar//net.urbanmc.ezauctions.acf.BaseCommand.executeCommand(BaseCommand.java:577)
[Thu, 2. Dec 2021 03:17:39 UTC WARN] at ezAuctions-1.6.4-SNAPSHOT.jar//net.urbanmc.ezauctions.acf.BaseCommand.execute(BaseCommand.java:513)
[Thu, 2. Dec 2021 03:17:39 UTC WARN] at ezAuctions-1.6.4-SNAPSHOT.jar//net.urbanmc.ezauctions.acf.RootCommand.execute(RootCommand.java:99)
[Thu, 2. Dec 2021 03:17:39 UTC WARN] at ezAuctions-1.6.4-SNAPSHOT.jar//net.urbanmc.ezauctions.acf.BukkitRootCommand.execute(BukkitRootCommand.java:81)
[Thu, 2. Dec 2021 03:17:39 UTC WARN] at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:159)
[Thu, 2. Dec 2021 03:17:39 UTC WARN] at org.bukkit.craftbukkit.v1_18_R1.CraftServer.dispatchCommand(CraftServer.java:878)
[Thu, 2. Dec 2021 03:17:39 UTC WARN] at net.minecraft.server.network.PlayerConnection.a(PlayerConnection.java:2201)
[Thu, 2. Dec 2021 03:17:44 UTC WARN] at net.minecraft.server.network.PlayerConnection.a(PlayerConnection.java:2012)
[Thu, 2. Dec 2021 03:17:44 UTC WARN] at net.minecraft.server.network.PlayerConnection.a(PlayerConnection.java:1993)
[Thu, 2. Dec 2021 03:17:44 UTC WARN] at net.minecraft.network.protocol.game.PacketPlayInChat.a(PacketPlayInChat.java:46)
[Thu, 2. Dec 2021 03:17:44 UTC WARN] at net.minecraft.network.protocol.game.PacketPlayInChat.a(PacketPlayInChat.java:6)
[Thu, 2. Dec 2021 03:17:44 UTC WARN] at net.minecraft.network.protocol.PlayerConnectionUtils.lambda$ensureRunningOnSameThread$1(PlayerConnectionUtils.java:56)
[Thu, 2. Dec 2021 03:17:44 UTC WARN] at net.minecraft.server.TickTask.run(TickTask.java:18)
[Thu, 2. Dec 2021 03:17:44 UTC WARN] at net.minecraft.util.thread.IAsyncTaskHandler.c(IAsyncTaskHandler.java:149)
[Thu, 2. Dec 2021 03:17:44 UTC WARN] at net.minecraft.util.thread.IAsyncTaskHandlerReentrant.c(IAsyncTaskHandlerReentrant.java:23)
[Thu, 2. Dec 2021 03:17:44 UTC WARN] at net.minecraft.server.MinecraftServer.b(MinecraftServer.java:1412)
[Thu, 2. Dec 2021 03:17:44 UTC WARN] at net.minecraft.server.MinecraftServer.c(MinecraftServer.java:189)
[Thu, 2. Dec 2021 03:17:44 UTC WARN] at net.minecraft.util.thread.IAsyncTaskHandler.y(IAsyncTaskHandler.java:122)
[Thu, 2. Dec 2021 03:17:44 UTC WARN] at net.minecraft.server.MinecraftServer.bf(MinecraftServer.java:1390)
[Thu, 2. Dec 2021 03:17:44 UTC WARN] at net.minecraft.server.MinecraftServer.y(MinecraftServer.java:1383)
[Thu, 2. Dec 2021 03:17:44 UTC WARN] at net.minecraft.util.thread.IAsyncTaskHandler.c(IAsyncTaskHandler.java:132)
[Thu, 2. Dec 2021 03:17:44 UTC WARN] at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1361)
[Thu, 2. Dec 2021 03:17:44 UTC WARN] at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1267)
[Thu, 2. Dec 2021 03:17:44 UTC WARN] at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:316)
[Thu, 2. Dec 2021 03:17:49 UTC WARN] at java.base/java.lang.Thread.run(Thread.java:833)
[Thu, 2. Dec 2021 03:17:49 UTC WARN] Error getting item as json. Item: BONE
[Thu, 2. Dec 2021 03:17:49 UTC INFO] [Auction] CamoMano is auctioning 1  for $1 at an increment of $1 for 60 seconds.
[Thu, 2. Dec 2021 03:17:49 UTC WARN] Error getting minecraft name for BONE
[Thu, 2. Dec 2021 03:17:49 UTC WARN] Error getting xp needed for repair for BONE
```
![auc](https://user-images.githubusercontent.com/43580782/144352298-566658b6-f7eb-481e-aff5-742d588e2f01.png)

**Comment by elian1203** [comment]
It seems that the whole NMS system has drastically changed in 1.18. I am looking into what can be done for the issue.

**Comment by elian1203** [comment]
Fixed in #19
